File: javascript/matrix/matrix.js
export class Matrix {
  constructor(rawMatrix) {
    this.rawMatrix = rawMatrix;
  }

  get rows() {
    return this.rawMatrix
      .split("\n")
      .map(rowAsString => rowAsString
        .split(' ')
        .map(c => Number.parseInt(c))
      );
  }

  get columns() {
    const columns = []
    for (let i = 0; i < this.rows[0].length; i++) {
      columns.push([]);
      this.rows.forEach(r => {
        columns[i].push(r[i]);
      });      
    }
    return columns;
  }
}

File: python/matrix/matrix.py
class Matrix:
    def __init__(self, matrix_string):
        rows_as_string = matrix_string.split("\n")
        self.rows = [list(map(lambda n: int(n), row_as_string.split(" "))) for row_as_string in rows_as_string]
        ''' I m not sure about readability of nested list comprehension'''
        self.columns = [[row[i] for row in self.rows] for i in range(len(self.rows[0]))]

    def row(self, index):
        return self.rows[index - 1]

    def column(self, index):
        return self.columns[index - 1]
